File: tests/fixtures/cpp/src/midi.cpp
#include "midi.h"
#include "globals.h"
#include <cstdio>

namespace midi {

bool note_on(uint8_t note, uint8_t velocity) {
    g_counter++;
    printf("[MIDI] NoteOn note=%d vel=%d\n", note, velocity);
    return note < 128 && velocity > 0;
}

bool control_change(uint8_t cc, uint8_t value) {
    g_counter++;
    return cc < 128;
}

std::vector<MidiMessage> generate_sequence(int length) {
    std::vector<MidiMessage> seq;
    seq.reserve(length);
    for (int i = 0; i < length; i++) {
        seq.push_back({0x90, static_cast<uint8_t>(60 + (i % 12)),
                        static_cast<uint8_t>(80 + (i % 40))});
    }
    return seq;
}

} // namespace midi

File: tests/fixtures/rust/src/midi.rs
pub fn note_on(note: u8, velocity: u8) -> bool {
    note < 128 && velocity > 0
}

pub fn control_change(cc: u8, value: u8) -> bool {
    let _ = value;
    cc < 128
}

pub fn generate_sequence(length: usize) -> Vec<(u8, u8)> {
    (0..length)
        .map(|i| (60 + (i as u8 % 12), 80 + (i as u8 % 40)))
        .collect()
}
